dali/python/nvidia/dali/types.py

```diff
@@ -157,74 +160,178 @@ def __init__(self, value, dtype=None):
-        return Constant(self.value, DALIDataType.FLOAT)
+        return ScalarConstant(self.value, DALIDataType.FLOAT)
 
     def float64(self):
-        return Constant(self.value, DALIDataType.FLOAT64)
+        return ScalarConstant(self.value, DALIDataType.FLOAT64)
 
     def __eq__(self, other):
-        if isinstance(other, Constant):
+        if isinstance(other, ScalarConstant):
             return self.value == other.value and self.dtype == other.dtype
         # Delegate the call to the `__eq__` of other object, most probably an `_EdgeReference`
         return other.__eq__(self)
 
     def __ne__(self, other):
-        if isinstance(other, Constant):
+        if isinstance(other, ScalarConstant):
             return self.value != other.value or self.dtype != other.dtype
         # Delegate the call to the `__ne__` of other object, most probably an `_EdgeReference`
         return other.__ne__(self)
 
     def __bool__(self):
         if self.dtype in _int_like_types:
             return bool(self.value)
-        raise TypeError(("DALI Constant must be converted to one of bool or int types: ({}) "
+        raise TypeError(("DALI ScalarConstant must be converted to one of bool or int types: ({}) "
                 "explicitly before casting to builtin `bool`.").format(_int_like_types))
 
     def __int__(self):
         if self.dtype in _int_like_types:
             return int(self.value)
-        raise TypeError(("DALI Constant must be converted to one of bool or int types: ({}) "
+        raise TypeError(("DALI ScalarConstant must be converted to one of bool or int types: ({}) "
                 "explicitly before casting to builtin `int`.").format(_int_like_types))
 
     def __float__(self):
         if self.dtype in _float_types:
             return self.value
-        raise TypeError(("DALI Constant must be converted to one of the float types: ({}) "
+        raise TypeError(("DALI ScalarConstant must be converted to one of the float types: ({}) "
                 "explicitly before casting to builtin `float`.").format(_float_types))
 
     def __str__(self):
         return "{}:{}".format(self.value, self.dtype)
 
     def __repr__(self):
         return "{}".format(self.value)
+
+def _is_scalar_shape(shape):
+    return shape is None or shape == 1 or shape == [1]
+
+def _is_numpy_array(value):
+    if 'numpy.ndarray' in str(type(value)):
+        return True
+
+def ConstantNode(device, value, dtype, shape, layout):
+    data = value
+    if _is_numpy_array(value):
+        import numpy as np
+
+        # 64-bit types are not supported - downgrade the input
+        if value.dtype == np.float64:
+            value = value.astype(np.float32)
+        if value.dtype == np.int64:
+            value = value.astype(np.int32)
+        if value.dtype == np.uint64:
+            value = value.astype(np.uint32)
+
+        def _numpy_to_dali_type(t):
+            if t is None:
+                return None
+            import numpy as np
```

Comment:
Do we need this import here?

Reply:
Not really, it's already there in the outer scope.